intermediate_source/char_rnn_classification_tutorial.py

```diff
@@ -164,9 +126,87 @@ def lineToTensor(line):
+
+print (f"The letter 'a' becomes {lineToTensor('a')}") #notice that the first position in the tensor = 1
+print (f"The name 'Ahn' becomes {lineToTensor('Ahn')}") #notice 'A' sets the 27th index to 1
 
-print(lineToTensor('Jones').size())
+#########################
+# Congratulations, you have built the foundational tensor objects for this learning task! You can use a similar approach 
+# for other RNN tasks with text.
+#
+# Next, we need to combine all our examples into a dataset so we can train, text and validate our models. For this, 
+# we will use the `Dataset and DataLoader <https://pytorch.org/tutorials/beginner/basics/data_tutorial.html>` classes 
+# to hold our dataset. Each Dataset needs to implement three functions: __init__, __len__, and __getitem__. 
+from io import open
+import glob
+import os
+import time 
+
+import torch
+from torch.utils.data import Dataset
+
+class NamesDataset(Dataset):
+
+    def __init__(self, data_dir):
+        self.data_dir = data_dir #for provenance of the dataset
+        self.load_time = time.localtime #for provenance of the dataset 
+        labels_set = set() #set of all classes
+
+        self.data = []
+        self.data_tensors = []
+        self.labels = [] 
+        self.labels_tensors = [] 
+
+        #read all the txt files in the specified directory
+        text_files = glob.glob(os.path.join(data_dir, '*.txt'))                           
+        for filename in text_files:
+            label = os.path.splitext(os.path.basename(filename))[0]
+            labels_set.add(label)
+            lines = open(filename, encoding='utf-8').read().strip().split('\n')
+            for name in lines: 
+                self.data.append(name)
+                self.data_tensors.append(lineToTensor(name))
+                self.labels.append(label)
+
+        #Cache the tensor representation of the labels 
+        self.labels_uniq = list(labels_set)
+        for idx in range(len(self.labels)):
+            temp_tensor = torch.tensor([self.labels_uniq.index(self.labels[idx])], dtype=torch.long)
+            self.labels_tensors.append(temp_tensor)
+
+    def __len__(self):
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        data_item = self.data[idx]
+        data_label = self.labels[idx]
+        data_tensor = self.data_tensors[idx]
+        label_tensor = self.labels_tensors[idx] 
+
+        return label_tensor, data_tensor, data_label, data_item
+
+
+#########################
+#Here we can load our example data into the NamesDataset
+
+alldata = NamesDataset("data/names")
+print(f"loaded {len(alldata)} items of data")
+print(f"example = {alldata[0]}")
+
+#########################
+#Using the dataset object allows us to easily split the data into train and test sets. Here we create a 80/20 
+#split but the torch.utils.data has more useful utilities. Here we specify a generator since we need to use the 
```

Comment:
Can you add a space between the initial # sign and the first letter of the character for consistency with other tutorials?
```suggestion
# split but the ``torch.utils.data`` has more useful utilities. Here we specify a generator since we need to use the 
```